doc/stdenv.xml

```diff
@@ -179,68 +179,285 @@ genericBuild
+  The platform distinctions are motivated by cross compilation; see <xref linkend="chap-cross"/> for exactly what each platform means.
+  <footnote><para>
+    The build platform is ignored because it is a mere implementation detail of the package satisfying the dependency:
+    As a general programming principle, dependencies are always <emphasis>specified</emphasis> as interfaces, not concrete implementation.
+  </para></footnote>
+  But even if one is not cross compiling, the platforms imply whether or not the dependency is needed at run-time or build-time, a concept that makes perfect sense outside of cross compilation.
+  For now, the run-time/build-time distinction is just a hint for mental clarity, but in the future it perhaps could be enforced.
+</para>
+<para>
+  The extension of <envar>PATH</envar> with dependencies, alluded to above, proceeds according to the relative platforms alone.
+  The process is carried only for dependencies whose host platform matches the new derivation's build platform–i.e. which runs on the directory where the new derivation will be built.
+  <footnote><para>Currently, that means all dependencies are put on the <envar>PATH</envar> for native builds, but in the future to match cross that may not be the case—the platforms would be assumed to be unique for consistency between cross and native builds.</para></footnote>
+  Of those, for each dependency <replaceable>dir</replaceable>, <filename><replaceable>dir</replaceable>/bin</filename>, if present, is added to the <envar>PATH</envar> environment variable.
+</para>
+<para>
+  The dependency is propagated when it forces some of its other-transitive (non-immediate) downstream dependencies to also take it on as an immediate dependency.
+  Nix itself already takes a package's transitive dependencies into account, but this propagation ensures nixpkgs-specific infrastructure like setup hooks (mentioned above) also are run as if the propagated dependency.
+</para>
+<para>
+  It is important to note dependencies are not necessary propagated as the same sort of dependency that they were before, but rather as the corresponding sort so that the platform rules still line up.
+  The exact rules for dependency propagation can be given by assigning each sort of dependency two integers based one how it's host and target platforms are offset from the depending derivation's platforms.
+  Those offsets are given are given below in the descriptions of each dependency list attribute.
+  Informally, we traverse propagated inputs, accumulating an dependency for each transitive path such that:
+  <itemizedlist>
+    <listitem>
+      <para>
+      The type of dependency is determined sum-like fold of the offsets assigned to each edge of the path
+      </para>
+    </listitem>
+    <listitem>
+      <para>
+      The path is only included if the subpaths sharing the starting point never go out of bounds (<literal>&lt; -2</literal>, <literal>&gt; 1</literal>).
```

Comment:
I'll admit I don't really understand either of these explanations. What do you mean by "subpaths sharing the starting point"?

Reply:
Yeah this description took a turn for the worse when I had to change it from an actual sum to something for complicated.

Reply:
Rewrote completely; should be clearer. It doesn't even attempt to be complete description now as only the formal part can really do that.